### PR TITLE
Test sensor2

### PR DESCRIPTION
**Pull request by Rithika-Rajan**
Test cases covered:

1. Test for invalid timestamp format.
2. Test for negative sensor values.
3. Test for successfully setting up all sensors.

Added error handling in the native device method in sensor file

### CODE REVIEW

File: homeassistant/components/openAQ/sensor.py [hunk 1]
```diff
@@ -154,11 +154,25 @@ def device_class(self):
...
+            if (
+                self.metric != SensorDeviceClass.TEMPERATURE
+                and data is not None
+                and data < 0
+            ):
+                return None
+
+            return data
+        except ValueError:
+            # Specific exception for handling datetime parsing errors
```
Comment (Ronserro) [inline homeassistant/components/openAQ/sensor.py]: if the exception handling is only for datetime parsing then it should not be general for all sensors

File: tests/components/openAQ/test_sensor.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    hass: HomeAssistant,
+    setup_integration: ComponentSetup,
+    config_entry: MockConfigEntry,
+):
+    """Test for successfully setting up all sensors."""
+    await setup_integration(
+        config_entry, "location_all_good.json", "measurements_all_good.json"
+    )
+    assert hass.states.get("sensor.temperature").state == "43.0"
+    assert hass.states.get("sensor.ozone").state
```
Comment (Ronserro) [inline tests/components/openAQ/test_sensor.py]: this should have a value as we are not testing init

File: tests/components/openAQ/test_sensor.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+@pytest.mark.asyncio
+async def test_invalid_timestamp_format(
+    hass: HomeAssistant,
+    setup_integration: ComponentSetup,
+    config_entry: MockConfigEntry,
+):
+    """Test for invalid timestamp format."""
+    await setup_integration(
+        config_entry,
+        "measurements_invalid_timestamp.json",
```
Comment (Ronserro) [inline tests/components/openAQ/test_sensor.py]: this is a location endpoint not measurement, keep the name convention the same for fixtures.

File: homeassistant/components/openAQ/sensor.py (hunk 1 above)
Comment (grace43) [reply homeassistant/components/openAQ/sensor.py]: Made it to be specific to datetime only

File: homeassistant/components/openAQ/sensor.py [hunk 4]
```diff
@@ -154,11 +154,26 @@ def device_class(self):
...
+                return data / 1000
+            if (
+                self.metric != SensorDeviceClass.TEMPERATURE
+                and data is not None
+                and data < 0
+            ):
+                return None
+
+            return data
+        except ValueError:
```
Comment (Ronserro) [inline homeassistant/components/openAQ/sensor.py]: this is dangerous, currently if the exception is caught and it is not SensorDeviceClass.TIMESTAMP the method will not return anything.

File: homeassistant/components/openAQ/sensor.py [hunk 5]
```diff
@@ -154,11 +154,26 @@ def device_class(self):
...
-                "%Y-%m-%dT%H:%M:%S%z",  # The datetime from the response will be a string that needs to be converted
-            )
-
-        return self.coordinator.data.get(name)
+        try:
+            name = self.entity_description.key
+            data = self.coordinator.data.get(name)
+
+            if self.metric == SensorDeviceClass.TIMESTAMP:
+                return datetime.strptime(
```
Comment (Ronserro) [inline homeassistant/components/openAQ/sensor.py]: try catch should be around this return statement

File: homeassistant/components/openAQ/sensor.py (hunk 4 above)
Comment (grace43) [reply homeassistant/components/openAQ/sensor.py]: Yes you are right fixing it

File: homeassistant/components/openAQ/sensor.py (hunk 5 above)
Comment (grace43) [reply homeassistant/components/openAQ/sensor.py]: Got it

**Review by Ronserro — APPROVED** [review]
lgtm

**Pull request merged**